src/internal_modules/roc_core/string_list.cpp

```diff
@@ -86,17 +128,26 @@ bool StringList::push_back(const char* begin, const char* end) {
     const size_t cur_sz = data_.size();
     const size_t add_sz = (size_t)(end - begin) + 1;
 
-    if (!grow_(cur_sz + add_sz)) {
+    const size_t align_head = AlignOps::align_as(sizeof(Header) + add_sz, 8);
+
+    Header str_header;
+    str_header.len = static_cast<uint32_t>(end - begin);
+    Footer str_footer;
+    str_footer.len = static_cast<uint32_t>(end - begin);
+
+    if (!grow_(cur_sz + align_head + sizeof(Footer))) {
         return false;
     }
 
-    if (!data_.resize(cur_sz + add_sz)) {
+    if (!data_.resize(cur_sz + align_head + sizeof(Footer))) {
         return false;
     }
 
     back_ = &data_[cur_sz];
-    memcpy(&data_[cur_sz], begin, add_sz - 1);
-    data_[cur_sz + add_sz - 1] = '\0';
+    memcpy(&data_[cur_sz], &str_header, sizeof(Header));              // copy header
+    memcpy(&data_[cur_sz + sizeof(Header)], begin, add_sz - 1);       // copy string
+    data_[cur_sz + sizeof(Header) + add_sz - 1] = '\0';               // add null
+    memcpy(&data_[cur_sz + align_head], &str_footer, sizeof(Footer)); // copy footer
```

Comment:
I think we can simplify this code a bit if we'll create pointers to header and footer and then write directly into them, instead of memcpy-ing header/footer. Like this:

```cpp
    const size_t cur_sz = data_.size();
    const size_t add_sz = sizeof(Header)
        + AlignOps::align_as(size_t(end - begin), sizeof(Header)) + sizeof(Footer);

    // resize ...

    Header* str_header = (Header*)&data_[cur_sz];
    Footer* str_footer = (Footer*)&data_[cur_sz + add_sz] - sizeof(Footer);

    // fill fields of str_header and str_footer ...
```